From 5d52be9a7c56e416fe98744f025e8b4aa92fd849 Mon Sep 17 00:00:00 2001
From: Umang Chheda <quic_uchheda@quicinc.com>
Date: Wed, 11 Oct 2023 20:57:16 +0530
Subject: [PATCH 1/2] PENDING: arm64: dts: qcm6490: remove voltage voting for
 USB rails for idp

USB driver does not vote for voltage on hsphy and ssphy
rails. Due to which the initial voltage set by bootloader
is overridden by regulator framework with min voltage specified
on regulator registration.

Fix this temporarily by removing voltage voting support, which
will prevent regulator framework overriding the voltage set by
bootloader.

This commit will be reverted once voltage voting support is added
in USB driver.

Signed-off-by: Umang Chheda <quic_uchheda@quicinc.com>
Signed-off-by: Salendarsingh Gaud <quic_sgaud@quicinc.com>
Signed-off-by: Atul Dhudase <quic_adhudase@quicinc.com>
Upstream-Status: Pending
---
 arch/arm64/boot/dts/qcom/qcm6490-idp.dts | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/qcm6490-idp.dts b/arch/arm64/boot/dts/qcom/qcm6490-idp.dts
index 14c73f55ea52..89e653c93ae8 100644
--- a/arch/arm64/boot/dts/qcom/qcm6490-idp.dts
+++ b/arch/arm64/boot/dts/qcom/qcm6490-idp.dts
@@ -519,6 +519,20 @@ &vreg_l9b_1p2 {
 	regulator-allowed-modes = <RPMH_REGULATOR_MODE_LPM RPMH_REGULATOR_MODE_HPM>;
 };
 
+&vreg_l1b_0p912 {
+	/delete-property/regulator-min-microvolt;
+	/delete-property/regulator-max-microvolt;
+	regulator-allow-set-load;
+	regulator-allowed-modes = <RPMH_REGULATOR_MODE_LPM RPMH_REGULATOR_MODE_HPM>;
+};
+
+&vreg_l10c_0p88 {
+	/delete-property/regulator-min-microvolt;
+	/delete-property/regulator-max-microvolt;
+	regulator-allow-set-load;
+	regulator-allowed-modes = <RPMH_REGULATOR_MODE_LPM RPMH_REGULATOR_MODE_HPM>;
+};
+
 &wifi {
 	memory-region = <&wlan_fw_mem>;
 };
-- 
2.25.1

